app/experimenter/nimbus-ui/src/components/App/ExperimentRoot/index.tsx

```diff
@@ -0,0 +1,163 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { navigate, RouteComponentProps, useMatch } from "@reach/router";
+import React, { useEffect, useRef } from "react";
+import { useAnalysis, useExperiment } from "../../../hooks";
+import { BASE_PATH, POLL_INTERVAL } from "../../../lib/constants";
+import { ExperimentContext } from "../../../lib/contexts";
+import { getStatus, StatusCheck } from "../../../lib/experiment";
+import { AnalysisData } from "../../../lib/visualization/types";
+import ApolloErrorAlert from "../../ApolloErrorAlert";
+import PageExperimentNotFound from "../../PageExperimentNotFound";
+import PageLoading from "../../PageLoading";
+
+export const ExperimentRoot = ({
+  // BASE_PATH is a constant in App, but some tests vary in RouterSlugProvider
+  basepath = BASE_PATH,
+  children,
+}: {
+  basepath?: string;
+  children: React.ReactNode;
+} & RouteComponentProps) => {
+  // Should always find a slug route match, here.
+  const { slug } = useMatch(`${basepath}/:slug/*`)!;
+
+  const {
+    experiment,
+    notFound,
+    loading,
+    startPolling,
+    stopPolling,
+    refetch,
+    error,
+  } = useExperiment(slug);
+
+  const status = getStatus(experiment);
+
+  const {
+    execute: fetchAnalysis,
+    result: analysis,
+    error: analysisError,
+    status: analysisFetchStatus,
+  } = useAnalysis();
+
+  const polling = useRef(false);
+  const analysisRequired = useRef(false);
+
+  const analysisLoading = analysisFetchStatus === "loading";
+  const analysisFetched = analysisFetchStatus !== "not-requested";
```

Comment:
Can we export these as constants?

Reply:
Could do, but probably not much point: they're a part of [the AsyncStateStatus type exported by react-async-hook](https://github.com/slorber/react-async-hook/blob/master/src/index.ts#L40-L44). VSCode will auto-complete these strings and TypeScript will validate them on compile.

Reply:
Oh interesting I see.  TS constant validation for the win here 🎉